bravado/mapping/spec.py

```diff
@@ -5,27 +6,39 @@
 from swagger_spec_validator import validator20
 
 from bravado.mapping.model import build_models
-from bravado.mapping.resource import build_resources
 from bravado.mapping.model import tag_models
 from bravado.mapping.model import fix_malformed_model_refs
+from bravado.mapping.resource import build_resources
 
 
 log = logging.getLogger(__name__)
 
 
-class Spec(object):
-    """A Swagger API Specification.
+CONFIG_DEFAULTS = {
+    # On the client side, validate incoming responses
+    # On the server side, validate outgoing responses
+    'validate_responses': True,
+}
+
 
-    :param spec_dict: Swagger API specification in json-like dict form
-    :param origin_url: URL from which the spec was retrieved.
-    :param http_client: :class:`bravado.http_client.HTTPClient`
+class Spec(object):
+    """Represents a Swagger Specification for a service.
     """
 
-    def __init__(self, spec_dict, origin_url=None, http_client=None):
+    def __init__(self, spec_dict, origin_url=None, http_client=None,
+                 config=None):
+        """
+        :param spec_dict: Swagger API specification in json-like dict form
+        :param origin_url: URL from which the spec was retrieved.
+        :param http_client: :class:`bravado.http_client.HTTPClient`
+        :param config: Configuration dict. See CONFIG_DEFAULTS.
+        """
         self.spec_dict = spec_dict
         self.origin_url = origin_url or 'unknown'
         self.http_client = http_client
         self.api_url = None
+        self.config = copy.deepcopy(CONFIG_DEFAULTS)
+        self.config.update(config or {})
```

Comment:
`deepcopy` is one of the worse performance offenders. We should never need to deepcopy.

Why not do this instead:

```python
self.config = dict(CONFIG_DEFAULTS, **(config or {}))
```

which is pythons version of "merge dicts"